subworkflows/local/shortread_adapterremoval.nf

```diff
@@ -24,63 +29,108 @@ workflow SHORTREAD_ADAPTERREMOVAL {
     ADAPTERREMOVAL_SINGLE ( ch_input_for_adapterremoval.single, [] )
     ADAPTERREMOVAL_PAIRED ( ch_input_for_adapterremoval.paired, [] )
 
-    // due to the slightly ugly output implementation of the current AdapterRemoval2 version, each file
-    // has to be exported in a separate channel, and we must manually recombine when necessary
+    /*
+     * Due to the ~slightly~ very ugly output implementation of the current AdapterRemoval2 version, each file
+     * has to be exported in a separate channel and we must manually recombine when necessary.
+     */
 
     if ( params.shortread_clipmerge_mergepairs && !params.shortread_clipmerge_excludeunmerged ) {
-        ch_adapterremoval_for_cat = ADAPTERREMOVAL_PAIRED.out.collapsed
-                                                .mix(
-                                                    ADAPTERREMOVAL_PAIRED.out.collapsed_truncated,
-                                                    ADAPTERREMOVAL_PAIRED.out.singles_truncated,
-                                                    ADAPTERREMOVAL_PAIRED.out.pair1_truncated,
-                                                    ADAPTERREMOVAL_PAIRED.out.pair2_truncated
-                                                    )
-                                                .map {
-                                                    meta, reads ->
-                                                        def meta_new = meta.clone()
-                                                        meta_new.single_end = true
-
-                                                        [ meta_new, reads ]
-                                                    }
-                                                    .groupTuple()
-
-        ch_adapterremoval_reads_prepped = CAT_FASTQ ( ch_adapterremoval_for_cat ).reads
-                                            .mix( ADAPTERREMOVAL_SINGLE.out.singles_truncated )
+
+        ENSURE_FASTQ_EXTENSION1(
+            Channel.empty().mix(
+                ADAPTERREMOVAL_PAIRED.out.collapsed,
+                ADAPTERREMOVAL_PAIRED.out.collapsed_truncated,
+                ADAPTERREMOVAL_PAIRED.out.singles_truncated,
+                ADAPTERREMOVAL_PAIRED.out.pair1_truncated,
+                ADAPTERREMOVAL_PAIRED.out.pair2_truncated
+            )
```

Comment:
I _personally_ don't like this testing of channel manipulation within module invocation (I find it harder to read), but it's fine for now. I wonder if we could try out hte Nextflow pipe thing... (if that applies for manipulations not just modules.)

Reply:
Not sure what you mean by 'testing'? I can certainly assign this to a channel first. I have mixed feelings about the pipe. With many transformations it makes things less legible in my opinion.

Reply:
I definitely prefer using the pipe in R ✌🏼